### PR TITLE
feat: big Mario

### PR DESCRIPTION
**Pull request by CatCoder36**
## Description
Implementation of the large Mario view and controls

## Related Issue(s)
none

## Changes Made

- Controls management refactoring
- Calculation of forces to move big mario

## Screenshots (if applicable)
image

## Test
For test the big Mario change the next files:
PlayerComponent line 31:
`State = PlayerState.BIG;`

EntityFactory line 90:
`AnimationComponent playerAnimationComponent = new AnimationComponent(Animations.GetAnimation(playerComponent.State), 64,128, 0.09f);`
Review: 
@isaias-rojas 
@aguirrekarina

### CODE REVIEW

**Comment by aguirrekarina** [comment]
image

image

The size of Mario crouching should be smaller in height than Mario standing, the crouching one is 64 x 88 and the standing one or the others are 64 x128

**Comment by GastonGutierrezC** [comment]
If Mario jumps under the first row of blocks, the jump is not noticeable. 


[Screencast from 26-06-24 13:59:40.webm](https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/135564826/70256cd3-d86a-49e2-82c6-0fda5eac49dc)

**Comment by aguirrekarina** [comment]
[8fe79463-f45f-4117-b376-8bd28b77d032.webm](https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/101665677/ef65fedb-79ba-4fb0-8f3a-1516bda04f0f)

Cuando mario esta cobre los bloques de arriba a veces no se puede mover

File: MarioGame/Source/Systems/PlayerMovementSystem.cs [hunk 1]
```diff
@@ -17,65 +18,60 @@ namespace SuperMarioBros.Source.Systems
...
                 var collider = player.GetComponent<ColliderComponent>();
                 var animation = player.GetComponent<AnimationComponent>();
                 var movement = player.GetComponent<MovementComponent>();
-                var keyboardState = Keyboard.GetState();
-                var gamePadState = GamePad.GetState(PlayerIndex.One);
-                var camera = player.GetComponent<CameraComponent>();
-                if (!player.GetComponent<PlayerComponent>().HasReachedEnd)
+                var input = player.GetComponent<InputComponent>();
+                var playerComponent = player.GetComponent<PlayerComponent>();
+                Console.WriteLine(animation.currentState + " :: curent state");
```
Comment (aguirrekarina) [inline MarioGame/Source/Systems/PlayerMovementSystem.cs]: Delete log

**Review by aguirrekarina — APPROVED** [review]

**Comment by GastonGutierrezC** [comment]
[Screencast from 03-07-24 17:56:11.webm](https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/135564826/77a34943-453d-4f20-9b11-3773059f0c18)

When Mario big wins and after lowering the flag, Mario stays there and does not move to the castle.

File: MarioGame/Source/Systems/PlayerMovementSystem.cs (hunk 1 above)
Comment (CatCoder36) [reply MarioGame/Source/Systems/PlayerMovementSystem.cs]: Solved

**Pull request closed without merging**